src/state/player.ts

```diff
@@ -14,169 +15,252 @@ export interface Timestamp {
 	href: string;
 }
 
-interface PlayerState {
-	current_show: null | Show;
-	playing: boolean;
-	currentTime: number;
-	audio?: HTMLAudioElement;
-	media_controller?: HTMLAudioElement;
-	status: 'INITIAL' | 'LOADED' | 'PAUSED' | 'PLAYING';
-}
-
 export const episode_share_status = writable<boolean>(false);
 
-const reset_state = {
-	playing: false,
-	status: 'INITIAL'
-} as const;
-
 const new_player_state = () => {
-	const player_state = writable<PlayerState>({
+	const initial_state: PlayerState = {
 		current_show: null,
-		audio: undefined,
-		media_controller: undefined,
-		currentTime: 0,
-		...reset_state
-	});
+		audio: null,
+		media_controller: null,
+		duration: 0,
+		status: 'INITIAL',
+		initial_load: true
+	};
 
+	const player_state = writable<PlayerState>(initial_state);
 	const { update, subscribe, set } = player_state;
 
-	async function start_show(requested_show: Show, start_time = 0) {
-		// First we check to see if the use has cached the mp3 offline
-		const show = await get_cached_or_network_show(requested_show);
-		const current_state = get(player_state);
-		if (show.url !== current_state?.audio?.src) {
-			return initialize_audio(show, start_time);
-		} else {
-			if (current_state.status === 'PLAYING') {
-				return pause();
-			} else if (current_state.status === 'PAUSED') {
-				return play();
-			} else {
-				return initialize_audio(show, start_time);
+	// An interval that saves time stamps to localstorage
+	let save_position_interval: number | null = null;
```

Comment:
Instead of setting an interval, can we just listen for all the events that would warrant an progress save? This is what I do on my platform.

timeupdate, play, pause, seeked, ended,

Reply:
timeupdate was the initial approach but it fires way too frequently, So I figured it would be better to control the interval we're saving.

Reply:
We could use timeupdate and throw a throttle on there for saving to storage

Reply:
Whats the harm in it firing so frequently? we arent sending any network requests, are we?

Reply:
Since it's saving to localstorage, I didn't want to use the memory by saving to storage several times a second. No harm other then that, just didn't think we needed the mem usage.

Reply:
ah yah, I dont think it would cause an issue, but we can throw a throttle around it if we need to

Reply:
Kk. update incoming.